charts/coturn/templates/certificate.yaml

```diff
@@ -0,0 +1,55 @@
+{{- if .Values.tls.enabled }}
+{{- if .Values.tls.certManager.enabled }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: coturn-auto-certificate
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  issuerRef:
+    name: {{ .Values.tls.certManager.issuer.name }}
+    kind: {{ .Values.tls.certManager.issuer.kind }}
+  usages:
+    - server auth
+  duration: 2160h # 90 days, ignored by LE
+  renewBefore: 360h # 15 days
+  isCA: false
+  secretName: coturn-tls
```

Comment:
This should definitely become a value (with `coturn-dns` as a default), so people not using cert-manager can specify the path to a secret resource containing a cert.

Reply:
What do you mean, "this should become a value"? I don't understand what you're referring to. Are you suggesting that the certificate parameters should be exposed in the chart configuration `values.yaml`?

Is it worth even referencing cert-manager resources in this chart at all, if all we need is the name of a `Secret` containing the certificates, irrespective of whether it's managed manually by the operator or buy cert-manager?

Reply:
Sorry, helm naming schemes are terrible. The idea was that instead of hardcoding `coturn-tls`, this parameter will become a helm value, or "variable".

And you're right, how these secrets are populated (either triggered by some cert-manager machinery via `Certificate` resources, or manually placed and rotated by an operator) should be entirely of scope for the helm chart. You just give it the name of the secret it tries to use, and that's it.

Reply:
Done in [3cae51b](https://github.com/wireapp/wire-server/pull/2620/commits/3cae51b9080c3c316e57fa93fe396418c348f4c9)